pandas/core/generic.py

```diff
@@ -8029,6 +8029,91 @@ def mask(self, cond, other=np.nan, inplace=False, axis=None, level=None,
         is not realigned. That is, use freq if you would like to extend the
         index when shifting and preserve the original data.
 
+        Examples
+        --------
+
+        >>> df = pd.DataFrame({'group': ['A', 'A', 'A', 'B', 'B', 'B'],
```

Comment:
This probably isn't enforceable and maybe it's just a personal preference but I think using `key` and `val` would be preferable to `group` and `myvalue`. I've seen the former used in quite a few more places so for consistency would be better to use those names.

@datapythonista maybe something to think about

Reply:
In general I think it makes things clearer if the example is with real world data. In this case, if we just show basic examples of `shift` (and we avoid using `ffill`, `reindexing`, `groupby` and `concat`), I don't think we need the column `group`.

I think the examples are quite good for the cookbook, or the time series documentation, but for the docstring, I think we should show the basic stuff. In this case what I'd do:
- Create a simple dataframe (one column and 4 rows for example)
- Show `.shift()` with default arguments
- Show how `period` can change the shifted periods
- Show how `freq` can be used

And I don't know exactly what `axis` does in this method, it could be shown too.

For all the rest of the stuff, feel free to contribute it to the other documents.